### DISCUSSION TITLE
filtering based on date not working

### DISCUSSION THREAD

**Discussion by ldobransky** [supabase-community/supabase-kt#690] [Q&A] [answered]
can somebody help me why filtering based on date not working in this case? I am sure that dates are not null, but response is without filtering. 
`
supabaseClient.postgrest["some_view"].select{
            
// Order by date descending 
order(column = "date", order = Order.DESCENDING)
            filter{
                // Filter by date range
                if (dateTo != null) {
                    lte("date", dateTo)
                }
                if (dateFrom != null) {
                    gte("date", dateFrom)
                }
            }
            // Apply pagination
            range(offset * (page - 1) until offset * page)
        }
`

**Comment by jan-tennert** [comment, marked as answer]
You can't directly apply multiple filters to one column, you have to use `and`:
```kotlin
and {
    if(dateTo != null) {
        lte("date", dateTo)
    }
    if(dateFrom != null) {
        gte("date", dateFrom)
    }
}
```
